inlang/source-code/sdk/src/loadProject.ts

```diff
@@ -36,24 +36,49 @@ const settingsCompiler = TypeCompiler.Compile(ProjectSettings)
  * Creates an inlang instance.
  *
  * @param projectPath - Absolute path to the inlang settings file.
- * @param nodeishFs - Filesystem that implements the NodeishFilesystemSubset interface.
+ * @param @deprecated nodeishFs - Filesystem that implements the NodeishFilesystemSubset interface.
  * @param _import - Use `_import` to pass a custom import function for testing,
  *   and supporting legacy resolvedModules such as CJS.
  * @param _capture - Use `_capture` to capture events for analytics.
  *
  */
-export const loadProject = async (args: {
+export async function loadProject(args: {
 	projectPath: string
-	repo?: Repository
-	nodeishFs: NodeishFilesystem
+	nodeishFs: Repository["nodeishFs"]
 	_import?: ImportFunction
 	_capture?: (id: string, props: Record<string, unknown>) => void
-}): Promise<InlangProject> => {
+}): Promise<InlangProject>
+
+/**
+ * @param projectPath - Absolute path to the inlang settings file.
+ * @param repo - An instance of a lix repo as returned by `openRepository`.
+ * @param _import - Use `_import` to pass a custom import function for testing,
+ *   and supporting legacy resolvedModules such as CJS.
+ * @param _capture - Use `_capture` to capture events for analytics.
+ *
+ */
+export async function loadProject(args: {
+	projectPath: string
+	repo: Repository
+	_import?: ImportFunction
+	_capture?: (id: string, props: Record<string, unknown>) => void
+}): Promise<InlangProject>
+
+export async function loadProject(args: any): Promise<InlangProject> {
 	const projectPath = normalizePath(args.projectPath)
 
+	let fs: Repository["nodeishFs"]
+	if (args.nodeishFs) {
+		fs = args.nodeishFs
+	} else if (args.repo) {
+		fs = args.repo.nodeishFs
+	} else {
+		throw new LoadProjectInvalidArgument(`Repo missing from arguments.`, { argument: "repo" })
+	}
+
 	// -- migrate if outdated ------------------------------------------------
 
-	await maybeMigrateToDirectory({ nodeishFs: args.nodeishFs, projectPath })
+	await maybeMigrateToDirectory({ nodeishFs: fs, projectPath })
```

Comment:
This place is a good opportunity to increase the readability by placing `await maybeCreateProjectId()` right beneath `maybeMigrateToDirectory`.

Reply:
the following code (from below) could be encapsulated in a function

```ts

		let projectId: string | undefined

		try {
			projectId = await nodeishFs.readFile(projectPath + "/project_id", {
				encoding: "utf-8",
			})
		} catch (error) {
			// @ts-ignore
			if (error.code === "ENOENT") {
				if (args.repo) {
					projectId = await generateProjectId(args.repo, projectPath)
					if (projectId) {
						await nodeishFs.writeFile(projectPath + "/project_id", projectId)
					}
				}
			} else {
				idError = error as Error
			}
		}


```